### PR TITLE
adjust AccountSettingsView

### PR DESCRIPTION
**Pull request by thienlnguyen**
- Users can update their profile information directly without requiring an additional button.
- Input fields now include validation.
- Mandatory fields are clearly marked.
- Changes can be discarded if needed.
- Unit tests using Vitest have been implemented

### CODE REVIEW

File: test/views/AccountSettingsView.spec.ts [hunk 1]
```diff
@@ -44,16 +47,87 @@ describe('AccountSettingsView', () => {
         countryCode: 'TC',
       },
     ]);
+    wrapper.vm.saveProfile = vi.fn();
+    wrapper.vm.fetchUserProfile = vi.fn();
+    wrapper.vm.validateAddress = vi.fn();
   });
 
   test('The view is rendered properly', () => {
     expect(wrapper.exists()).toBe(true);
```
Comment (AliceKitchkin) [inline test/views/AccountSettingsView.spec.ts:56]: ### Hinweise für Thien
Der Test scheint noch unvollständig zu sein. Sollte die Vervollständigung innerhalb des Issue-Scopes liegen, müsste der Test noch ergänzt werden.

File: test/views/AccountSettingsView.spec.ts
Comment (AliceKitchkin) [inline test/views/AccountSettingsView.spec.ts:1]: Jeder Test sollte eine kurze Beschreibung als Kommentar bekommen. Die hier hinzugefügten Tests decken ein paar Edge Cases ab, jedoch noch nicht weitere mögliche Szenarien. Je nach Issue-Scope, sollte das mit dem Dozenten geklärt und die Tests eventuell erweitert werden.

**Review by AliceKitchkin — CHANGES_REQUESTED** [review]
### Review des Pull Requests #356 
Alle Hinweise und Notizen wurden, für eine bessere Verständlichkeit, in den genauen Codezeilen hinterlegt. Dabei gibt es die Abschnitte "Hinweise für Thien", welche Vorschläge für Änderungen enthalten, und "Notizen", welcher zur Verständlichkeit des Codes für den Reviewer beitagen soll.

**Ansicht vorher:**
Benutzerprofil_Bearbeiten_leer

**Ansicht nachher:**
Benutzerprofil_Ansicht

### Allgemeine Anmerkungen
**Code Kommentare:**
Für die schnelle Verständlichkeit und bessere Lesbarkeit fehlen im Code Kommentare, diese sollten sinnvoll und einheitlich in Englisch hinterlegt werden.

**Commit Messages:**
Vor dem Merge wird im Regelfall nur der "Diff" betrachtet, dennoch sollten die Commit Messages in Zukunft etwas ausführlicher dokumentiert werden.

**Testabdeckung:**
Durch SonarCloud ergab sich eine Testabdeckung von 77,8%, 80% sind erforderlich. Es könnten folgende Tests ergänzt werden:
- Testen, ob das Speichern des Profils verhindert wird, wenn eine ungültige Telefonnummer eingegeben wird
- Testen, ob die Erfolgs- und Fehler-Pop-Ups korrekt angezeigt werden, wenn das Speichern erfolgreich ist oder fehlschlägt

**Test Mocking:**
Die Methoden werden direkt auf `wrapper.vm.$options` gemockt. Eventuell sollten diese Methoden lieber auf der Instanz `wrapper.vm` gemockt werden, um sicherzustellen, dass die gemockten Methoden tatsächlich aufgerufen werden.
```suggestion
    vi.spyOn(wrapper.vm, 'fetchUserProfile').mockResolvedValue({ ...
```

**Rebase:**
Achte darauf, ein Rebase durchzuführen, bevor du ein PR erstellst. Dadurch vermeidest du irgendwelche Merge-Konflikte, weil du auf dem neuesten Stand bist.

### Zu erfüllende Anforderungen
Im Issue wurden folgende Anforderungen gelistet und erfüllt/nicht erfüllt:
- [x] Users can update their profile information directly without requiring an additional button. _-> Erfüllt: Benutzer gelangt direkt zur Editoransicht_
- [ ] Input fields now include validation. _-> Teilweise erfüllt: Fehlermeldungen werden teilweise nicht korrekt ein- und ausgeblendet. Siehe Review in Codezeilen_
- [X] Mandatory fields are clearly marked. _-> Erfüllt: Schwarze Sternchen_
- [X] Changes can be discarded if needed. _-> Erfüllt: Abbrechen-Button_
- [X] Unit tests using Vitest have been implemented _-> Erfüllt: Tests laufen durch ohne Error_

### Ergänzende Anforderungen
Es haben sich beim Review ein paar (weitere) Kann-Anforderungen ergeben.

**Nice to have: Logik**
- E-Mail ändern können
- Eingabeüberprüfung von Stadt und Bundesland (z.B. auf Korrektheit oder keine Zahlen)
- Eingabeüberprüfung der Adresse, sollte mindestens eine Zahl enthalten für die Hausnummer

**Nice to have: Design**
- Pflichtfelder-Sternchen rot markieren statt schwarz
- Abstand zwischen den Buttons "Speichern" und "Abbrechen" einügen
- Abstand zwischen allen Buttons und der horizontalen Linie einfügen
Abstand_Buttons

**Comment by AliceKitchkin** [comment]
@thienlnguyen please read my review and the requested changes :)

File: test/views/AccountSettingsView.spec.ts (hunk 1 above)
Comment (thienlnguyen) [reply test/views/AccountSettingsView.spec.ts:56]: Hi Alice,
die von dir markierten Codezeilen im beforeEach-Block dienen dazu, die Methoden saveProfile(), fetchUserProfile() und validateAddress() zu mocken. Die Mocks ermöglichen die Methoden später in den spezifischen Tests zu überwachen.

Die Tests für diese Funktionen sind bereits an den folgenden Stellen implementiert: saveProfile() in Zeile 96, fetchUserProfile() in Zeile 59 und validateAddress() in Zeile 116.

File: src/views/AccountSettingsView.vue [hunk 2]
```diff
@@ -72,337 +74,464 @@ async function fetchUserProfile() {
...
+  return typeof value === 'string' ? value : '';
+}
+
+function getUpdatedAddressValue(field: keyof Address): string {
+  const value = editedAddress.value?.[field] || addressProfile.value?.[field];
+  return typeof value === 'string' ? value : '';
+}
+
+async function saveProfile(): Promise<void> {
+  try {
```
[the remark this answers is not in the document]
Comment (thienlnguyen) [reply src/views/AccountSettingsView.vue:99]: Hi Alice, 
die Felder werden bereits validiert, wenn Änderungen vorgenommen werden. Ich habe den Code nun so angepasst, dass der Speichern-Button disabled wird, sobald Fehlermeldungen erscheinen.

File: src/views/AccountSettingsView.vue [hunk 3]
```diff
@@ -72,337 +74,464 @@ async function fetchUserProfile() {
...
+
+    const updatedUser = await userService.updateUser(user);
+    console.log('Benutzer erfolgreich aktualisiert:', updatedUser);
+  } catch (e) {
+    console.error('Das Benutzerprofil konnte nicht geupdated werden!', e);
+    alert('Fehler beim Aktualisieren des Benutzerprofils!');
   }
 }
 
+function validateAddress(address: Partial<Address>): boolean {
```
[the remark this answers is not in the document]
Comment (AliceKitchkin) [reply src/views/AccountSettingsView.vue:136]: - [x] done

File: src/views/AccountSettingsView.vue [hunk 4]
```diff
@@ -72,337 +74,464 @@ async function fetchUserProfile() {
...
+    editedAddress.value.countryCode = matchingCountry.code; // Kein Fehler
+  }
 }
 
 async function getCity() {
   const userService = new UserService();
-  const address = await userService.getCityFromZip(editedUserProfile.value.address.zip);
+  const address = await userService.getCityFromZip(editedAddress.value.zip);
   try {
+    errorMessage.value.zip = 'Bitte eingeben!';
```
[the remark this answers is not in the document]
Comment (AliceKitchkin) [reply src/views/AccountSettingsView.vue:234]: - [x] done

File: test/views/AccountSettingsView.spec.ts
Comment (AliceKitchkin) [reply test/views/AccountSettingsView.spec.ts:1]: - [x] done

File: test/views/AccountSettingsView.spec.ts (hunk 1 above)
Comment (AliceKitchkin) [reply test/views/AccountSettingsView.spec.ts:56]: - [x] done

File: src/views/AccountSettingsView.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (AliceKitchkin) [reply src/views/AccountSettingsView.vue:102]: - [x] done

**Review by AliceKitchkin — APPROVED** [review]

**Comment by AliceKitchkin** [comment]
@astanik please take a look, the code is ready to merge for me 👍
